Understanding Lightning Network Implementation Architecture (#36)
peakshift/bitcoin-ux: payments/lightning-architecture.md

# Lightning Network App Infrastructure

> Over the course of the development of the Lighting Network over the years, there have been a number of different attempts to port LN functionality to mobile devices and end users.

Reviewer: What is meant by end-users?

```suggestion
Over the years as the Lighting Network developed, there have been a number of different attempts to port it's functionality to mobile devices.
```

> The traditional way to interact with the Lightning Network was to run a Bitcoin full node with a Lightning Network node attached to it on a dedicated device and to interact with it either via the command line or a web-based UX that attaches directly on top of the node. This approach is very technical, and was only viable in the early days of development, or now with technical folks who work around the protocol or who don't mind working through the extra steps for thismore robust setup.

Reviewer: ```suggestion
The traditional way to interact with the Lightning Network was to run a Bitcoin full node with a Lightning Network node attached to it on a dedicated device. The command-line or RPC would be used to interact with the node, and sometimes a web-based tool that connects via the available APIs would be used.

This approach is very technical and was only viable in the early days of development, or now with technical folks who work around the protocol or who don't mind working through the extra steps for a more "robust" setup.
```

---
**_(Incomplete)_**
## Mapping the Lightning Layers Model to LDK Architecture

A mapping of how the distinct libraries in theLDK correspond to the different components in the Lighting protocol layers model.

### LDK Non-Core Components (all batteries)

These include mostly components for interacting with the Bitcoin network level of the architecture. It mostly has to do with managing and storing UTXOs/keys. The two Lightning Network related components in this category are defining:
    - Disk I/O: how channels get stored (local storage, cloud etc)
    - TCP networking: how messages get sent along to LN peers

**Bitcoin Network based**
- **Blockchain data** (feeds into ChainMonitor)
    e.g. from Neutrino

- **Wallet**
    for on-chain UTXO management logic (display balances, create txns etc.)

- **Tx Broadcast** (Bitcoin Networking)
    for broadcasting signed txns to the Bitcoin network

- **Private key management**
    for signing txns

- **Fee estimation**
    for monitoring the mempool and suggesting sensible fees

> **Lightning Infrastructure based**

Reviewer: @vindard what is this?

Reviewer: nvm i got it now
